Loader

idoc07 opened the pull request.

idoc07 committed:
Loader

idoc07 committed:
LoaderConstants

idoc07 committed:
Update Loader.java

idoc07 committed:
Update LoaderConstants.java

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,30 @@
+public class Loader extends SubsystemBase {
+    private WPI_TalonSRX motor;
+
+    private LoaderConstants.LoaderState currentState;
```
static import

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    private Loader() {
+        motor = LoaderConstants.MOTOR;
```
move initialization to definition

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public void setState(LoaderConstants.LoaderState state) {
+        motor.set(ControlMode.PercentOutput, state.voltage / LoaderConstants.VOLTAGE_COMPENSATION);
```
set `currentState`...

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+
+public class LoaderConstants {
+    private static final int MOTOR_ID = 0;
+    protected static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
```
move to bottom

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+
+public class LoaderConstants {
+    private static final int MOTOR_ID = 0;
+    protected static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
+    protected static final double
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
+    protected static final double VOLTAGE_COMPENSATION = 10.5;
```
change all protected fields to default

idoc07 committed:
Static import to LoaderState, emptied constractor, fixed currentState, changed protected constant to default

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    public void setState(LoaderState state) {
+        motor.set(ControlMode.PercentOutput, state.voltage / LoaderConstants.VOLTAGE_COMPENSATION);
```
`ControlMode.PercentOutput` is unnecessary as that is the default

lycaon7 replied:
`.set` takes a percent between 1 and -1 not a voltage. there is a `.setVoltage` however unless @laviRZ told you to use voltage and not a percent then voltage is unnecessary here.

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
+    static final double VOLTAGE_COMPENSATION = 10.5;
+    static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
```
put above the voltages under the id

idoc07 committed:
changed according to the comments

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,31 @@
+import static frc.trigon.robot.subsystems.loader.LoaderConstants.LoaderState;
+
+public class Loader extends SubsystemBase {
+    private WPI_TalonSRX motor = LoaderConstants.MOTOR;
```
final

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+public class LoaderConstants {
+    private static final int MOTOR_ID = 0;
+    static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
+    static final double
```
private

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+
+public class LoaderConstants {
+    private static final int MOTOR_ID = 0;
+    static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
```
move to the bottom of the constants

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,24 @@
+    static final double
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
+    static final double VOLTAGE_COMPENSATION = 10.5;
```
voltage compensation is not enabled

idoc07 committed:
According to the first three reviews

idoc07 committed:
enabling voltage compensation

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,29 @@
+
+public class LoaderConstants {
+    private static final double
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
```
This should be power not voltage

idoc07 committed:
changing voltage to power

idoc07 committed:
reformated

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,30 @@
+    private LoaderState currentState;
+
+    private Loader() {
+
```
extra newline

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public void setState(LoaderState state) {
+        motor.set(state.power / LoaderConstants.VOLTAGE_COMPENSATION);
```
extract to a function inside `utilities.Conversions` called `voltageToCompensatedPower`

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,29 @@
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+public class LoaderConstants {
+    static final double VOLTAGE_COMPENSATION = 10.5;
```
rename to voltage compensation saturation

laviRZ replied:
add an inversion flag

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,29 @@
+public class LoaderConstants {
+    static final double VOLTAGE_COMPENSATION = 10.5;
+    private static final double
+            LOAD_POWER = 6,
+            EJECT_POWER = -6;
```
this is voltage, not power

idoc07 committed:
according to the 6 rewiews (look at the description)

Opening a Conversation document, changing voltage to power, adding a boolean called IS_INVERTED, etc.

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,31 @@
+    private static final double
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
+    private static final boolean IS_INVERTED = false;
```
rename to motor inverted

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,9 @@
+package frc.trigon.robot.utilities;
+
+public class Conversions {
+    public static final double VOLTAGE_SATURATION = 10.5;
```
rename to voltage compensation saturation

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,9 @@
+public class Conversions {
+    public static final double VOLTAGE_SATURATION = 10.5;
+
+    public static double voltageToCompensatedVoltage(double voltage) {
```
to compensated power, not compensated voltage

idoc07 committed:
IS_INVERTED -> MOTOR_INVERTED, VOLTAGE_SATURATION -> VOLTAGE_COMPENSATION_SATURATION, voltageToCompensatedVoltage -> voltageToCompensatedPower. Documentation

idoc07 committed:
fixing the documentation

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    /**
+     * Gets the state of the Loader
```
redundant

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Gets the state of the Loader
+     *
+     * @return the current LoaderState
```
The current state of the loader. There's no need to specify the return type in the doc.

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Sets the state of the Loader
+     *
+     * @param state the wanted LoaderState
```
don't use the type of the parameter. "the wanted state for the loader" is a good description

idoc07 committed:
fixing the documentation

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,15 @@
+package frc.trigon.robot.utilities;
+
+public class Conversions {
+    public static final double VOLTAGE_COMPENSATION_SATURATION = 10.5;
```
this is motor dependent

idoc07 committed:
Setting VOLTAGE_COMPENSATION_SATURATION in LoaderConstants and not in Conversions

idoc07 committed:
documentation

idoc07 committed:
editing the documentation in the Conversions document

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+    /**
+     * Converts voltage to compensated power
+     *
+     * Talon FX/SRX and Victor SPX can be configured to adjust their outputs in response to the battery
+     * voltage measurement (in all control modes). Use the voltage compensation saturation config to
+     * determine what voltage represents 100% output.
```
I don't want this copypasta here

idoc07 replied:
bruh

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+     * Talon FX/SRX and Victor SPX can be configured to adjust their outputs in response to the battery
+     * voltage measurement (in all control modes). Use the voltage compensation saturation config to
+     * determine what voltage represents 100% output.
+     * The compensated power is the voltage divided by the saturation of the compensation
```
I don't want the calculation, I want to know what saturation and compensation means.

idoc07 replied:
bruh

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,38 @@
+    private Loader() {
+    }
+
+    public static Loader getInstance() {
```
Add Java Docs

shay124578 replied:
he doesn't need to do that

idoc07 replied:
do I need or not?

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,32 @@
+    private static final double
+            LOAD_VOLTAGE = 6,
+            EJECT_VOLTAGE = -6;
+    public static final double VOLTAGE_COMPENSATION_SATURATION = 10.5;
```
This is global. put the compensation as constant in conversions

lycaon7 commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,32 @@
+
+    public enum LoaderState {
+        LOAD(LOAD_VOLTAGE),
+        OFF(0),
```
put `OFF(0)` below eject

idoc07 committed:
putting off below eject

YU NO MAN BROKO MI

idoc07 committed:
what the compensated power is

For more info CLICK HERE

idoc07 committed:
what the voltage compensation saturation is

I'm done with this

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    /**
+     * Sets the state of the loader
```
Dot at the end

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/Loader.java:
```diff
@@ -0,0 +1,38 @@
+     * @param state the wanted state for the loader
+     */
+    public void setState(LoaderState state) {
+        motor.set(Conversions.voltageToCompensatedPower(state.voltage, LoaderConstants.VOLTAGE_COMPENSATION_SATURATION));
```
store in a variable before this line

idoc07 replied:
done

laviRZ commented on src/main/java/frc/trigon/robot/subsystems/loader/LoaderConstants.java:
```diff
@@ -0,0 +1,32 @@
+    static final WPI_TalonSRX MOTOR = new WPI_TalonSRX(MOTOR_ID);
+
+    static {
+        MOTOR.setInverted(MOTOR_INVERTED);
```
separate the two blocks of code here with a newline

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+public class Conversions {
+
+    /**
+     * Converts voltage to compensated power
```
Dot at the end

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+    /**
+     * Converts voltage to compensated power
+     *
+     * The voltage compensation saturation will determine what voltage represents 100% output.
+     *
+     * The compensated power is the power resulting from turning the voltage off and on without stopping.
+     * It's basically the voltage divided by the saturation of the compensation.
```
Not good enough
You need to explain in general the point of voltage compensation.
Then for each parameter explain its meaning.

idoc07 replied:
IDK

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+     * The compensated power is the power resulting from turning the voltage off and on without stopping.
+     * It's basically the voltage divided by the saturation of the compensation.
+     *
+     * @param voltage the voltage
```
the voltage of what

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+     * It's basically the voltage divided by the saturation of the compensation.
+     *
+     * @param voltage the voltage
+     * @param voltageCompensationSaturation the saturation of the compensation
```
the saturation of the compensation?

idoc07 replied:
yes

laviRZ commented on src/main/java/frc/trigon/robot/utilities/Conversions.java:
```diff
@@ -0,0 +1,20 @@
+     *
+     * @param voltage the voltage
+     * @param voltageCompensationSaturation the saturation of the compensation
+     * @return the compensated power
```
compensated for what?

idoc07 replied:
Any problem???

idoc07 committed:
adding dots

idoc07 committed:
editing documentation

idoc07 commented:
I'm retiring, goodbye.

idoc07 committed:
adding notes

Guys, if you see this, I'm not really retiring right now.

idoc07 committed:
adding notes

idoc07 committed:
storing in a variable

idoc07 committed:
separating the two blocks of code with a newline
